libs/backend/src/cast_vote_records/export.ts

```diff
@@ -839,14 +850,17 @@ export async function doesUsbDriveRequireCastVoteRecordSync(
     if (ballotsCounted === 0) {
       return false;
     }
-    const castVoteRecordRootHash = scannerStore.getCastVoteRecordRootHash();
 
     // A previous export operation may have failed midway
     if (areOrWereCastVoteRecordsBeingExportedToUsbDrive(usbDriveStatus)) {
       return true;
     }
 
     const { election, electionHash } = electionDefinition;
+    const exportDirectoryName = scannerStore.getExportDirectoryName();
+    if (!exportDirectoryName) {
+      return true;
+    }
```

Comment:
In case you're wondering why this check has moved location and gone from checking `false` to `true`, this check, at its previous location, was almost equivalent to the `ballotsCounted === 0` check (since we create the export directory after the first ballot is counted).

Now, by moving it after the `ballotsCounted === 0` case, we're catching the case that we fail to create the export directory (and specifically save its path in the DB) after the first ballot is counted. Should be exceedingly rare but, regardless, the new check is more useful than the previous one.

If you want to get extra specific, technically the `areOrWereCastVoteRecordsBeingExportedToUsbDrive(usbDriveStatus)` would likely also catch this. But that check involves a marker file on the USB drive. If the USB drive were replaced with a blank USB drive, that check wouldn't catch anything (other checks after would).

Zooming out, we're talking about the edgiest of edge cases here